### ISSUE TITLE
Master becomes locked in distributed mode  

### ISSUE THREAD

**Issue by artbez** [orientechnologies/orientdb#8306]
### OS: Ubuntu 16.04.2

We start two nodes in distributed mode. One master and one replica. Initially, the system behaves normally, but after a while the master is blocked and write operations are not executed.

## Expected behavior  
All write operations are performed on the master and then sent to the replica. A replica creates a full backup every 3 hours.  

## Actual behavior  
Write operations are not executed after the following lines from the master log.

```
2018-06-04 12:57:19:048 INFO  [master]->[replica1] Creating delta backup of database 'test' (startLSN=OLogSequenceNumber{segment=5, position=52}) in directory: /tmp/orientdb/backup_replica1_test_server0.zip... 
2018-06-04 12:57:19:049 INFO  Cannot find requested LSN=OLogSequenceNumber{segment=5, position=52} for database sync operation (last available LSN is OLogSequenceNumber{segment=21, position=214419})
2018-06-04 12:57:19:050 FINE  [master]->[replica1] Sending response com.orientechnologies.orient.server.distributed.task.ODistributedDatabaseDeltaSyncException: Requested database delta sync with LSN=OLogSequenceNumber{segment=5, position=52} but not found in database back (reqId=id=1. 26236 task=deploy_delta_db)
```
In replica log the master has backup status, but it should not do backups.

```
2018-06-04 12:57:19:189 WARNI [replica1]<-[master] Error on installing database delta for 'test' (err=Requested database delta sync with LSN=OLogSequenceNumber{segment=5, position=52} but not found in database)
2018-06-04 12:57:19:189 INFO  [replica1] Requesting full sync for database 'test'...
2018-06-04 12:57:19:189 INFO  [replica1]->[[master]] Requesting deploy of database 'test' on local server...
2018-06-04 12:57:19:206 INFO  [replica1]<-[master] Received updated status master.test=BACKUP
2018-06-04 12:57:19:207 INFO  [replica1] Distributed servers status (*=current @=lockmgr[master]):
```

## Steps to reproduce

Replica has following config.

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<orient-server>
    <handlers>
        <handler class="com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin">
            <parameters>
                <parameter value="${distributed}" name="enabled"/>
                <parameter value="${ORIENTDB_HOME}/config/default-distributed-db-config.json" name="configuration.db.default"/>
                <parameter value="${ORIENTDB_HOME}/config/hazelcast.xml" name="configuration.hazelcast"/>
                <parameter value="replica1" name="nodeName"/>
            </parameters>
        </handler>
        ...
        <handler class="com.orientechnologies.orient.server.handler.OAutomaticBackup">
            <parameters>
                <parameter value="true" name="enabled"/>
                <parameter value="${ORIENTDB_HOME}/config/automatic-backup.json" name="config"/>
            </parameters>
        </handler>
        ...
    </orient-server>
</handlers>
```

Master has following.
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<orient-server>
    <handlers>
        <handler class="com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin">
            <parameters>
                <parameter value="${distributed}" name="enabled"/>
                <parameter value="${ORIENTDB_HOME}/config/default-distributed-db-config.json" name="configuration.db.default"/>
                <parameter value="${ORIENTDB_HOME}/config/hazelcast.xml" name="configuration.hazelcast"/>
                <parameter value="master" name="nodeName"/>
            </parameters>
        </handler>
        ...
        <handler class="com.orientechnologies.orient.server.handler.OAutomaticBackup">
            <parameters>
                <parameter value="false" name="enabled"/>
                <parameter value="${ORIENTDB_HOME}/config/automatic-backup.json" name="config"/>
            </parameters>
        </handler>
        ....
    </handlers>
</orient-server>
```
Distributed configuration is the same for both.
```
{
  "autoDeploy": true,
  "readQuorum": 1,
  "writeQuorum": "majority",
  "executionMode": "undefined",
  "readYourWrites": true,
  "newNodeStrategy": "dynamic",
  "servers": {
    "master": "master",
    "*": "replica"
  },
  "clusters": {
    "internal": {
    },
    "*": {
      "servers": ["<NEW_NODE>"]
    }
  }
}
```

**Comment by tglman** [comment]
Hi @artbez,

We fixed some similar issue in 3.0.2 that is beening released now, anyway could you share a thread dump so we can be sure about your specific problem.

Regards

**Comment by tglman** [comment]
Hi @artbez,

There is no public issue of that, anyway this is the commit that solve the problem: https://github.com/orientechnologies/orientdb/commit/71edf2df4b8bde6f47dd3d7293ea4028dcfc1c39

Regards
